### PR TITLE
[SNAP-2242] Unique application names & kill app by names

### PR DESCRIPTION
**Pull request by rishitesh**
## What changes were proposed in this pull request?
The standalone cluster should support unique application names. As they are user visible and easy to track user can write scripts to kill applications by names.
Also, I have added support to kill Spark applications by names(case insensitive).

## How was this patch tested?

Manual & precheckin

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/deploy/master/ui/MasterPage.scala [hunk 1]
```diff
@@ -67,6 +76,17 @@ private[ui] class MasterPage(parent: MasterWebUI) extends WebUIPage("") {
     }
   }
 
+  private def handleStopRequest(request: HttpServletRequest, action: String => Unit): Unit = {
+    if (parent.killEnabled &&
+        parent.master.securityMgr.checkModifyPermissions(request.getRemoteUser)) {
+      val name = Option(request.getParameter("name"))
```
Comment (sumwale) [inline core/src/main/scala/org/apache/spark/deploy/master/ui/MasterPage.scala]: Don't you need to check for "terminate" parameter like done in handleKillRequest?

File: core/src/main/scala/org/apache/spark/deploy/master/ui/MasterWebUI.scala [hunk 2]
```diff
@@ -50,6 +50,8 @@ class MasterWebUI(
     attachHandler(createStaticHandler(MasterWebUI.STATIC_RESOURCE_DIR, "/static"))
     attachHandler(createRedirectHandler(
       "/app/kill", "/", masterPage.handleAppKillRequest, httpMethods = Set("POST")))
+    attachHandler(createRedirectHandler(
+      "/app/stop", "/", masterPage.handleAppStopRequest, httpMethods = Set("POST")))
```
Comment (sumwale) [inline core/src/main/scala/org/apache/spark/deploy/master/ui/MasterWebUI.scala]: Better to call this "/app/killByName". Method names too.

File: core/src/test/scala/org/apache/spark/deploy/master/ui/MasterWebUISuite.scala [hunk 3]
```diff
@@ -74,6 +74,24 @@ class MasterWebUISuite extends SparkFunSuite with BeforeAndAfterAll {
     verify(master, times(1)).removeApplication(activeApp, ApplicationState.KILLED)
   }
 
+  test("Stop application") {
```
Comment (sumwale) [inline core/src/test/scala/org/apache/spark/deploy/master/ui/MasterWebUISuite.scala]: "Kill application by name"

File: core/src/main/scala/org/apache/spark/deploy/master/ui/MasterPage.scala (hunk 1 above)
Comment (sumwale) [reply core/src/main/scala/org/apache/spark/deploy/master/ui/MasterPage.scala]: Better to instead add a parameter to handleKillRequest itself (like "byName: Boolean") and change the "id" to use getParameter("name") for that case.

**Review by sumwale — APPROVED** [review]
Some comments, looks good otherwise. Please try upstream contribution especially because it is for spark standalone cluster only.

**Comment by rishitesh** [comment]
All review comments incorporated.

**Pull request merged**